### ISSUE TITLE
Partition creation isn't transactional

### ISSUE THREAD

**Issue by saaros** [postgrespro/pg_pathman#20]
The partition cache in shared memory is apparently updated immediately when partitions are updated, but that's not safe as other transactions won't see the new tables until they are committed.  And there's nothing that would refresh the cache in case a transaction is rolled back, so tables are essentially broken until you restart postgres in case a partition modifying transaction is rolled back:

```
os=# begin;
BEGIN
os=# SELECT create_range_partitions('kala', 'id', 0, 50000, 3);

[......] 

create_range_partitions 
-------------------------
3
os=# rollback;
ROLLBACK
os=# select * from kala;
ERROR:  could not open relation with OID 16556
```

**Comment by funbringer** [comment]
Hi again, thanks for pointing this out. There's still a few unsolved issues like this one, we're aware and working on a solution. I'll keep you informed about the progress.

**Comment by funbringer** [comment]
After a brief discussion we'd come up with an idea. Ildar Musin has just pushed a commit `9a57913` which seems to fix this annoying bug.

**Comment by funbringer** [comment]
@saaros 

I had to make a few improvements actually, so you're welcome to test them.

**Comment by funbringer** [comment]
This issue has been resolved in 1.0_rc1, so I'm closing it. Please let us know if anything doesn't work as expected.
